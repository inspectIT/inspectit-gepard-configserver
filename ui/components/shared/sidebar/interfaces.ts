import { IconType } from "react-icons";

/*
NavItem is an interface that defines the structure of the sidebar navigation items.
The Icon is provided as a react-icon component, and the color is optional.
*/
export interface NavItem {
  title: string;
  href: string;
  icon: IconType;
  color?: string;
  isChidren?: boolean;
  children?: NavItem[];
}
