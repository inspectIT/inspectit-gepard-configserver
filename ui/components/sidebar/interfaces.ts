import { IconType } from "react-icons";

export interface NavItem {
  title: string;
  href: string;
  icon: IconType;
  color?: string;
  isChidren?: boolean;
  children?: NavItem[];
}
