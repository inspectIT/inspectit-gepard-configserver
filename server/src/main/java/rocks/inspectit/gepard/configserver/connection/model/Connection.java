package rocks.inspectit.gepard.configserver.connection.model;

import jakarta.persistence.*;
import lombok.*;
import rocks.inspectit.gepard.configserver.agent.Agent;

import java.util.UUID;

@Entity
@Table
@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class Connection {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private UUID id;

    @Embedded
    private Agent agent;
}
