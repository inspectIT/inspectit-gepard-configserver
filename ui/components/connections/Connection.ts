import { UUID } from "crypto";
import { Agent } from "./Agent";

export interface Connection {
  id: UUID;
  agent: Agent;
}
