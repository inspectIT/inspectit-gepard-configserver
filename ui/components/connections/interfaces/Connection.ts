import { UUID } from "crypto";

export interface Connection {
  id: UUID;
  pid: number;
  serviceName: string;
  otelVersion: string;
  gepardVersion: string;
  startTime: string;
  javaVersion: string;
}
