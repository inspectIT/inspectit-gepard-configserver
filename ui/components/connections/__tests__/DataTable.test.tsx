import { beforeEach, describe, expect, it, test, vi } from "vitest";
import { cleanup, render, screen } from "@testing-library/react";
import Page from "@/app/page";

import Providers from "@/app/providers";

import { SetStateAction } from "react";
import { Connection } from "../interfaces/Connection";
import DataTable from "@/components/shared/shadcn/DataTable";
import { TestConnectionsBig } from "./TestConnections";

const setup = (data: Connection[]) => {
  return render(
    <Providers>
      <DataTable
        columns={[]}
        data={[]}
        globalFilter={""}
        setGlobalFilter={function (value: SetStateAction<string>): void {
          throw new Error("Function not implemented.");
        }}
      />
    </Providers>
  );
};

describe("Data Table No Data", () => {
  let containerWithoutData: HTMLElement;

  // Arrange
  beforeEach(() => {
    cleanup();
    const renderedDom = setup([]);
    containerWithoutData = renderedDom.container;
  });

  it("renders successfully", async () => {
    expect(containerWithoutData).toMatchSnapshot();
  });

  it('renders "no results" without data', async () => {
    const text = await screen.findByText("No results.");
  });
});

describe("Data Table with Data", () => {
  let containerWithData: HTMLElement;
  beforeEach(() => {
    const renderedDom = setup(TestConnectionsBig);
    containerWithData = renderedDom.container;
  });

  it("renders result with connections data", async () => {
    expect(containerWithData).toMatchSnapshot();
  });
});
