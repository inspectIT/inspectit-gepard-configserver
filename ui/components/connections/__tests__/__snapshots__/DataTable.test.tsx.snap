// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ConnectionsPage No Data > renders successfully 1`] = `
<div>
  <script>
    !function(){try{var d=document.documentElement,c=d.classList;c.remove('light','dark');var e=localStorage.getItem('theme');if('system'===e||(!e&&true)){var t='(prefers-color-scheme: dark)',m=window.matchMedia(t);if(m.media!==t||m.matches){d.style.colorScheme = 'dark';c.add('dark')}else{d.style.colorScheme = 'light';c.add('light')}}else if(e){c.add(e|| '')}if(e==='light'||e==='dark')d.style.colorScheme=e}catch(e){}}()
  </script>
  <div
    class="relative overflow-hidden sm:h-[60vh] 2xl:h-full"
    dir="ltr"
    style="position: relative; --radix-scroll-area-corner-width: 0px; --radix-scroll-area-corner-height: 0px;"
  >
    <style>
      [data-radix-scroll-area-viewport]{scrollbar-width:none;-ms-overflow-style:none;-webkit-overflow-scrolling:touch;}[data-radix-scroll-area-viewport]::-webkit-scrollbar{display:none}
    </style>
    <div
      class="h-full w-full rounded-[inherit]"
      data-radix-scroll-area-viewport=""
      style="overflow-x: scroll; overflow-y: scroll;"
    >
      <div
        style="min-width: 100%; display: table;"
      >
        <table
          class="w-full caption-bottom text-sm border"
        >
          <thead
            class="[&_tr]:border-b sticky top-0 bg-secondary"
          >
            <tr
              class="border-b transition-colors hover:bg-muted/50 data-[state=selected]:bg-muted"
            />
          </thead>
          <tbody
            class="[&_tr:last-child]:border-0"
          >
            <tr
              class="border-b transition-colors hover:bg-muted/50 data-[state=selected]:bg-muted"
            >
              <td
                class="p-4 align-middle [&:has([role=checkbox])]:pr-0 h-24 text-center"
                colspan="0"
              >
                No results.
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>
  <div
    class="flex items-center justify-end space-x-2 py-4"
  >
    <button
      class="inline-flex items-center justify-center whitespace-nowrap text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 border border-input bg-background hover:bg-accent hover:text-accent-foreground h-9 rounded-md px-3"
      disabled=""
    >
      Previous
    </button>
    <button
      class="inline-flex items-center justify-center whitespace-nowrap text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 border border-input bg-background hover:bg-accent hover:text-accent-foreground h-9 rounded-md px-3"
      disabled=""
    >
      Next
    </button>
  </div>
</div>
`;

exports[`ConnectionsPage with Data > renders result with connections data 1`] = `
<div>
  <script>
    !function(){try{var d=document.documentElement,c=d.classList;c.remove('light','dark');var e=localStorage.getItem('theme');if('system'===e||(!e&&true)){var t='(prefers-color-scheme: dark)',m=window.matchMedia(t);if(m.media!==t||m.matches){d.style.colorScheme = 'dark';c.add('dark')}else{d.style.colorScheme = 'light';c.add('light')}}else if(e){c.add(e|| '')}if(e==='light'||e==='dark')d.style.colorScheme=e}catch(e){}}()
  </script>
  <div
    class="relative overflow-hidden sm:h-[60vh] 2xl:h-full"
    dir="ltr"
    style="position: relative; --radix-scroll-area-corner-width: 0px; --radix-scroll-area-corner-height: 0px;"
  >
    <style>
      [data-radix-scroll-area-viewport]{scrollbar-width:none;-ms-overflow-style:none;-webkit-overflow-scrolling:touch;}[data-radix-scroll-area-viewport]::-webkit-scrollbar{display:none}
    </style>
    <div
      class="h-full w-full rounded-[inherit]"
      data-radix-scroll-area-viewport=""
      style="overflow-x: scroll; overflow-y: scroll;"
    >
      <div
        style="min-width: 100%; display: table;"
      >
        <table
          class="w-full caption-bottom text-sm border"
        >
          <thead
            class="[&_tr]:border-b sticky top-0 bg-secondary"
          >
            <tr
              class="border-b transition-colors hover:bg-muted/50 data-[state=selected]:bg-muted"
            />
          </thead>
          <tbody
            class="[&_tr:last-child]:border-0"
          >
            <tr
              class="border-b transition-colors hover:bg-muted/50 data-[state=selected]:bg-muted"
            >
              <td
                class="p-4 align-middle [&:has([role=checkbox])]:pr-0 h-24 text-center"
                colspan="0"
              >
                No results.
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>
  <div
    class="flex items-center justify-end space-x-2 py-4"
  >
    <button
      class="inline-flex items-center justify-center whitespace-nowrap text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 border border-input bg-background hover:bg-accent hover:text-accent-foreground h-9 rounded-md px-3"
      disabled=""
    >
      Previous
    </button>
    <button
      class="inline-flex items-center justify-center whitespace-nowrap text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 border border-input bg-background hover:bg-accent hover:text-accent-foreground h-9 rounded-md px-3"
      disabled=""
    >
      Next
    </button>
  </div>
</div>
`;

exports[`Data Table No Data > renders successfully 1`] = `
<div>
  <script>
    !function(){try{var d=document.documentElement,c=d.classList;c.remove('light','dark');var e=localStorage.getItem('theme');if('system'===e||(!e&&true)){var t='(prefers-color-scheme: dark)',m=window.matchMedia(t);if(m.media!==t||m.matches){d.style.colorScheme = 'dark';c.add('dark')}else{d.style.colorScheme = 'light';c.add('light')}}else if(e){c.add(e|| '')}if(e==='light'||e==='dark')d.style.colorScheme=e}catch(e){}}()
  </script>
  <div
    class="relative overflow-hidden sm:h-[60vh] 2xl:h-full"
    dir="ltr"
    style="position: relative; --radix-scroll-area-corner-width: 0px; --radix-scroll-area-corner-height: 0px;"
  >
    <style>
      [data-radix-scroll-area-viewport]{scrollbar-width:none;-ms-overflow-style:none;-webkit-overflow-scrolling:touch;}[data-radix-scroll-area-viewport]::-webkit-scrollbar{display:none}
    </style>
    <div
      class="h-full w-full rounded-[inherit]"
      data-radix-scroll-area-viewport=""
      style="overflow-x: scroll; overflow-y: scroll;"
    >
      <div
        style="min-width: 100%; display: table;"
      >
        <table
          class="w-full caption-bottom text-sm border"
        >
          <thead
            class="[&_tr]:border-b sticky top-0 bg-secondary"
          >
            <tr
              class="border-b transition-colors hover:bg-muted/50 data-[state=selected]:bg-muted"
            />
          </thead>
          <tbody
            class="[&_tr:last-child]:border-0"
          >
            <tr
              class="border-b transition-colors hover:bg-muted/50 data-[state=selected]:bg-muted"
            >
              <td
                class="p-4 align-middle [&:has([role=checkbox])]:pr-0 h-24 text-center"
                colspan="0"
              >
                No results.
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>
  <div
    class="flex items-center justify-end space-x-2 py-4"
  >
    <button
      class="inline-flex items-center justify-center whitespace-nowrap text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 border border-input bg-background hover:bg-accent hover:text-accent-foreground h-9 rounded-md px-3"
      disabled=""
    >
      Previous
    </button>
    <button
      class="inline-flex items-center justify-center whitespace-nowrap text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 border border-input bg-background hover:bg-accent hover:text-accent-foreground h-9 rounded-md px-3"
      disabled=""
    >
      Next
    </button>
  </div>
</div>
`;

exports[`Data Table with Data > renders result with connections data 1`] = `
<div>
  <script>
    !function(){try{var d=document.documentElement,c=d.classList;c.remove('light','dark');var e=localStorage.getItem('theme');if('system'===e||(!e&&true)){var t='(prefers-color-scheme: dark)',m=window.matchMedia(t);if(m.media!==t||m.matches){d.style.colorScheme = 'dark';c.add('dark')}else{d.style.colorScheme = 'light';c.add('light')}}else if(e){c.add(e|| '')}if(e==='light'||e==='dark')d.style.colorScheme=e}catch(e){}}()
  </script>
  <div
    class="relative overflow-hidden sm:h-[60vh] 2xl:h-full"
    dir="ltr"
    style="position: relative; --radix-scroll-area-corner-width: 0px; --radix-scroll-area-corner-height: 0px;"
  >
    <style>
      [data-radix-scroll-area-viewport]{scrollbar-width:none;-ms-overflow-style:none;-webkit-overflow-scrolling:touch;}[data-radix-scroll-area-viewport]::-webkit-scrollbar{display:none}
    </style>
    <div
      class="h-full w-full rounded-[inherit]"
      data-radix-scroll-area-viewport=""
      style="overflow-x: scroll; overflow-y: scroll;"
    >
      <div
        style="min-width: 100%; display: table;"
      >
        <table
          class="w-full caption-bottom text-sm border"
        >
          <thead
            class="[&_tr]:border-b sticky top-0 bg-secondary"
          >
            <tr
              class="border-b transition-colors hover:bg-muted/50 data-[state=selected]:bg-muted"
            />
          </thead>
          <tbody
            class="[&_tr:last-child]:border-0"
          >
            <tr
              class="border-b transition-colors hover:bg-muted/50 data-[state=selected]:bg-muted"
            >
              <td
                class="p-4 align-middle [&:has([role=checkbox])]:pr-0 h-24 text-center"
                colspan="0"
              >
                No results.
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>
  <div
    class="flex items-center justify-end space-x-2 py-4"
  >
    <button
      class="inline-flex items-center justify-center whitespace-nowrap text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 border border-input bg-background hover:bg-accent hover:text-accent-foreground h-9 rounded-md px-3"
      disabled=""
    >
      Previous
    </button>
    <button
      class="inline-flex items-center justify-center whitespace-nowrap text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 border border-input bg-background hover:bg-accent hover:text-accent-foreground h-9 rounded-md px-3"
      disabled=""
    >
      Next
    </button>
  </div>
</div>
`;
